html/includes/common/availability-map.inc.php

```diff
@@ -12,41 +12,58 @@
  * the source code distribution for details.
  */
 
+$sql = dbFetchRow('SELECT `settings` FROM `users_widgets` WHERE `user_id` = ? AND `widget_id` = ?', array($_SESSION["user_id"], '1'));
+$widget_mode = json_decode($sql['settings']);
```

Comment:
@crcro This query creates the potential for unexpected behavior (i.e. if you have this widget on multiple dashboards, with different settings, you don't know which one you will get) and defeats the dashboard specific query for widget settings that is already being done.  Which also means you don't know which dashboard will control the direct page view.  Can we rework this so the widget and direct view are independent entities?